pkg/controller/controller.go

```diff
@@ -560,29 +560,23 @@ func (c *M3DBController) handleClusterUpdate(cluster *myspec.M3DBCluster) error
 			return err
 		}
 
-		if !update {
-			continue
+		var processNext bool
+		// The expectation is that these update methods will return false when they have updated
+		// actual pods in the statefulset. This will stop any further processing of pods until
+		// the updated ones are back healthy
+		if cluster.Spec.OnDeleteUpdateStrategy {
```

Comment:
Do we know if we were to update a `StatefulSet`'s `template` and its `updateStrategy` if the controller will use the new `updateStrategy` to update the set or the old one? Just wondering about a potential edge case where we update both, and the StatefulSet controller used the old strategy for the update but because we're checking `cluster.Spec.OnDeleteUpdateStrategy` here we expect the new one will be used. I would _think_ the controller would use the new strategy that's in the update but Kube has surprised me before 😄

Reply:
Yeah, so I just tested this out and it does respect the new `updateStrategy`. So when you change the template and set `OnDelete`, k8s (correctly, imo) waits for you to roll the statefulset pods to pick up the update.